libs/sdk-ui-dashboard/src/presentation/topBar/menuButton/DefaultMenuButton.tsx

```diff
@@ -26,24 +28,151 @@ const bubbleAlignPoints: IAlignPoint[] = [{ align: "cl tr" }];
 export const DefaultMenuButton = (props: IMenuButtonProps): JSX.Element | null => {
     const { menuItems } = props;
     const [isOpen, setIsOpen] = useState(false);
+    const [autofocusSubmenu, setAutofocusSubmenu] = useState(false);
     const intl = useIntl();
     const tooltipText = intl.formatMessage({ id: "controlButtons.options.tooltip" });
+    const backLabel = intl.formatMessage({ id: "controlButtons.options.back" });
+    const closeLabel = intl.formatMessage({ id: "controlButtons.options.close" });
+    const menuWrapperRef = useRef<HTMLDivElement>(null!);
+    const [parentItemId, setParentItemId] = useState<string | null>(null);
+    const menuItemRefs = useRef<Map<string, HTMLElement>>(new Map());
 
-    // generate unique anchor class name to open dropdown next to the correct button if app uses multiple
-    // dashboard components
-    const dropdownAnchorClassName = useMemo(() => `dash-header-options-anchor-${uuid()}`, []);
+    const setMenuItemRef = useCallback(
+        (itemId: string) => (element: HTMLDivElement | HTMLButtonElement | null) => {
+            if (element) {
+                menuItemRefs.current.set(itemId, element);
+            } else {
+                menuItemRefs.current.delete(itemId);
+            }
+        },
+        [],
+    );
+
+    // generate unique IDs for accessibility and dropdown positioning
+    const idPrefix = useId();
+    const buttonId = `${idPrefix}-menu-button`;
+    const menuId = `${idPrefix}-menu`;
```

Comment:
Does it need to have shared ID part?

Reply:
I prefer it to easily find it by search in DOM view. I did the same with dropdownButton+dropdownBody